src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1260,21 +1260,24 @@ internal virtual void FillRequestStream(HttpRequestMessage request)
 
             foreach (var entry in WebSession.ContentHeaders)
             {
-                if (SkipHeaderValidation)
-                {
-                    request.Content.Headers.TryAddWithoutValidation(entry.Key, entry.Value);
-                }
-                else
+                if (!string.IsNullOrWhiteSpace(entry.Value))
                 {
-                    try
+                    if (SkipHeaderValidation)
```

Comment:
If our intention is to follow Core (that is to send headers with empty value) the check should be here:
```suggestion
                    if (SkipHeaderValidation || entry.Value?.Length == 0)
```

Also please add a test.